--- orig/u-boot-marvell.git/include/configs/openrd_base.h	2011-05-05 15:44:14.000000000 +0900
+++ u-boot-marvell.git/include/configs/openrd_base.h	2011-05-06 12:29:08.000000000 +0900
@@ -33,7 +33,7 @@
 /*
  * Version number information
  */
-#define CONFIG_IDENT_STRING	"\nOpenRD_base"
+#define CONFIG_IDENT_STRING	"\nOpenRD"
 
 /*
  * High Level Configuration Options (easy to change)
@@ -53,6 +53,7 @@
 #define CONFIG_CMD_DHCP
 #define CONFIG_CMD_ENV
 #define CONFIG_CMD_MII
+#define CONFIG_CMD_MMC
 #define CONFIG_CMD_NAND
 #define CONFIG_CMD_PING
 #define CONFIG_CMD_USB
@@ -110,6 +111,14 @@
 #endif /* CONFIG_CMD_NET */
 
 /*
+ * SDIO/MMC Card Configuration
+ */
+#ifdef CONFIG_CMD_MMC
+#define CONFIG_MMC
+#define CONFIG_MV_SDIO
+#endif /* CONFIG_CMD_MMC */
+
+/*
  * SATA Driver configuration
  */
 #ifdef CONFIG_MVSATA_IDE
@@ -117,4 +126,10 @@
 #define CONFIG_SYS_ATA_IDE1_OFFSET	MV_SATA_PORT1_OFFSET
 #endif /*CONFIG_MVSATA_IDE*/
 
+/*
+ * File system
+ */
+#define CONFIG_CMD_EXT2
+#define CONFIG_CMD_FAT
+
 #endif /* _CONFIG_OPENRD_BASE_H */
