--- agmodinst-dma.c.orig	2016-12-23 20:56:52.000000000 +0900
+++ agmodinst-dma.c	2019-04-28 09:33:49.785639607 +0900
@@ -101,6 +101,8 @@
     down_read(&current->mm->mmap_sem);
 #if LINUX_VERSION_CODE < KERNEL_VERSION(4,6,1)
     dma->nbr_pages = get_user_pages(current, current->mm, addr, nbr_pages_needed, 1, 0, dma->pages, NULL);
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(4,9,0)
+    dma->nbr_pages = get_user_pages(addr, nbr_pages_needed, 0, dma->pages, NULL);
 #else
     dma->nbr_pages = get_user_pages(addr, nbr_pages_needed, 1, 0, dma->pages, NULL);
 #endif
